src/compression.rs

```diff
@@ -1,26 +1,14 @@
 use anyhow::{Context, Result};
 
 pub(crate) fn decompress_value(v: &[u8]) -> Result<Vec<u8>> {
-    if v.is_empty() {
-        return Ok(vec![]);
-    }
-
-    let mut out_len = v.len() * 6;
-    loop {
-        match lz4_flex::block::decompress(v, out_len) {
-            Ok(out) => {
-                return Ok(out);
-            }
-            Err(lz4_flex::block::DecompressError::OutputTooSmall { expected, actual }) => {
-                assert_eq!(actual, out_len);
-                assert!(
-                    expected > actual,
-                    "failed: expected ({expected}) > actual ({actual})"
-                );
-                out_len = expected.max(actual.saturating_mul(2));
-            }
-            Err(e) => {
-                return Err(e).context("decompression");
+    match lz4_flex::block::decompress(v, v.len() * 6) {
+        Ok(result) => Ok(result),
+        Err(e) => {
+            // If it failed and the error suggests buffer too small, retry with 10x
+            if e.to_string().contains("too small") {
```

Comment:
Scanning the stringified error is a hack. I guess you're getting an error variant that is not `OutputTooSmall`? If so, that sounds like an upstream bug to me.

Reply:
Actually I don't see why you fix would work, because the only error variant that can result in `too small` is the one that the existing code already handles. This variant is NEVER returned. So how did your binary produce that error message? Are you sure you're using the latest `main` commit?

Reply:
Fair enough. I am still learning Rust but this patch fixed the issue for me. No need to merge just wanted it out there for others if it might help. Thank you for your time!

Reply:
Well maybe there's a bug and I just don't understand it. Could you retry the unpatched version and run `cargo run -- --version`? I'm asking because the last bug report for that particular code routine got fixed by #230 and if lz4 freaks out, I would expect one of the assertions to fire.